torch/_dynamo/allowed_functions.py

```diff
@@ -54,40 +54,71 @@ def make_function_id_set(lazy_initializer):
     added to the graph and what will cause a graph break.
     """
 
-    class FunctionIdSet:
-        function_ids: Optional[Set[int]] = None
-        function_names: Optional[Dict[int, str]] = None
-
-        def __call__(self):
-            if self.function_ids is None:
-                value = lazy_initializer()
-                if isinstance(value, dict):
-                    self.function_ids = set(value.keys())
-                    self.function_names = value
+    _pending_functions: List[Tuple["FunctionIdSet", bool, int]] = []
+    _function_ids: Optional[Set[int]] = None
+    _function_names: Optional[Dict[int, str]] = None
```

Comment:
Won't these be shared across all instances of FunctionIdSet since they are defined on the class?

I worry about mutations to `_pending_functions` leaking across object instances.